compile_state/src/state.rs

```diff
@@ -480,9 +480,29 @@ Create a proper list from pairs with items 0 - N.
 Section: pair
 
 Example:
-(test::assert-equal '(1 2 3) (list 1 2 3))"),
-            list_append: add_special(vm, "list-append", ""),
-            cons: add_special(vm, "cons", ""),
+(test::assert-equal '(1 2 3) (list 1 2 3))
+(test::assert-not-equal (list-append (list 1 2 3) (list 1)) (list-append (list 1 2 3) 1))
+"),
+            list_append: add_special(vm, "list-append", r#"Usage: (list-append list item)
+
+If second parameter is a list it will be appened to the first list, otherwise
+appends item to list.
+
+Section: pair
+
+Example:
+(test::assert-equal (list-append (list 1 2 3) (list 1)) (list 1 2 3 1))
+(test::assert-not-equal (list-append (list 1 2 3) (list 1)) (list-append (list 1 2 3) 1))
```

Comment:
Maybe add a note or expand docs slightly to cover why these are not equal.  On first blush it looks like they would be...